hr_holidays_public/models/hr_holidays.py

```diff
@@ -0,0 +1,103 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from dateutil import tz
+import pytz
+from odoo import api, fields, models
+
+
+def to_naive_user_tz(date_time, record):
+    tz_name = record.env.context.get('tz') or record.env.user.tz
+    tz = pytz.timezone(tz_name) if tz_name else pytz.UTC
+    return pytz.UTC.localize(
+        date_time.replace(tzinfo=None), is_dst=False,
+    ).astimezone(tz).replace(tzinfo=None)
+
+
+class HrHolidays(models.Model):
+    _inherit = 'hr.holidays'
+
+    show_full_days = fields.Boolean(
+        string="Show Full Days",
+        related="holiday_status_id.compute_full_days")
+    number_of_hours_temp = fields.Float(
+        'Allocated Hours', copy=False, readonly=True,
+        states={'draft': [('readonly', False)],
+                'confirm': [('readonly', False)]},
+        help='Number of hours of the leave request according '
+             'to your working schedule.')
+    number_of_hours = fields.Float(
+        'Number of Hours', compute='_compute_number_of_hours',
+        store=True, track_visibility='onchange')
+
+    @api.multi
+    @api.depends('number_of_hours_temp', 'type')
+    def _compute_number_of_hours(self):
+        for holiday in self:
+            if holiday.type == 'remove':
+                holiday.number_of_hours = -holiday.number_of_hours_temp
+            else:
+                holiday.number_of_hours = holiday.number_of_hours_temp
+
+    @api.onchange('employee_id')
+    def _onchange_employee(self):
+        super(HrHolidays, self)._onchange_employee()
+        self._onchange_data_hr_holidays_compute_days()
+
+    @api.onchange('date_from')
+    def _onchange_date_from(self):
+        if not self.show_full_days:
+            super(HrHolidays, self)._onchange_date_from()
+        if self.show_full_days and self.date_from:
+            tz_name = self.employee_id.user_id.tz or self.env.user.tz
+            start_dt = to_naive_user_tz(
+                fields.Datetime.from_string(self.date_from), self)
+            dt = start_dt.replace(
+                hour=0, minute=0, second=0, microsecond=0,
+                tzinfo=tz.gettz(tz_name),
+            ).astimezone(tz.tzutc())
+            self.date_from = fields.Datetime.to_string(dt)
+            if self.date_to:
+                end_dt = to_naive_user_tz(
+                    fields.Datetime.from_string(self.date_to), self)
+                dt = end_dt.replace(
+                    hour=23, minute=59, second=59, microsecond=999999,
+                    tzinfo=tz.gettz(tz_name),
+                ).astimezone(tz.tzutc())
+                self.date_to = fields.Datetime.to_string(dt)
+        self._onchange_data_hr_holidays_compute_days()
+
+    @api.onchange('date_to')
+    def _onchange_date_to(self):
+        super(HrHolidays, self)._onchange_date_to()
+        if self.show_full_days and self.date_to:
+            tz_name = self.employee_id.user_id.tz or self.env.user.tz
```

Comment:
`tz_name` still can be zero. Can it make the error?

Reply:
If not set, dateutil gets the tz from os...